### DISCUSSION TITLE
Could HOOMD-Blue do the affine deformation?

### DISCUSSION THREAD

**Discussion by maojinyuan** [glotzerlab/hoomd-blue#1855] [Support] [answered]
Hi there,

I am currently using HOOMD version 2.9.7 to perform deformation simulations.

While working on uniaxial stretching, I couldn't find the arguments for square root interpolation. For example:
`update.box_resize(Lx = hoomd.variant.linear_interp([(0, Lx0), (1e6, Lx1)]),
                  Ly = hoomd.variant.linear_interp([(0, Ly0), (1e6, Ly1)]),
                  Lz = hoomd.variant.linear_interp([(0, Lz0), (1e6, Lz1)]))`

When resizing the box with Lx changing from `Lx0` to `Lx1` using linear_interp, the values for `Ly1` and `Lz1` should be `Ly0/sqrt(2)` and `Lz0/sqrt(2)`, respectively. Am I missing any arguments to accomplish this?

Thanks in advance!

**Comment by mphoward** [comment, marked as answer]
You can define a `CustomVariant` in currently supported versions of HOOMD:

https://hoomd-blue.readthedocs.io/en/v4.8.1/module-hoomd-variant.html#hoomd.variant.Variant

Otherwise, you can add more points to your linear interpolation to approximate a square root.

**Reply by joaander** [reply]
You'll want to use a custom [`BoxVariant`](https://hoomd-blue.readthedocs.io/en/v4.8.1/module-hoomd-variant-box.html)
